community/cypher/cypher-compiler-3.2/src/test/scala/org/neo4j/cypher/internal/compiler/v3_2/planner/logical/LeafPlanningIntegrationTest.scala

```diff
@@ -723,4 +724,95 @@ class LeafPlanningIntegrationTest extends CypherFunSuite with LogicalPlanningTes
     plan should equal(distinct)
   }
 
+  //---------------------------------------------------------------------------
+  // Test expand order with multiple configurations and
+  // unsupported.cypher.plan_with_minimum_cardinality_estimates setting
+  //
+  // To succeed this test assumes:
+  // *  (:A) should have lower cardinality than (:B) and (:C) so it is selected as starting point
+  // * (a)--(b) should have lower cardinality than (a)--(c) so that it is expanded first
+  //
+  // Ideally (and at the time of writing) the intrinsic order when the cardinalities are equal
+  // is different from the assertion and would cause failure
+  private def testAndAssertExpandOrder(config: StubbedLogicalPlanningConfiguration) {
+    val query = "MATCH (b:B)-[rB]->(a:A)<-[rC]-(c:C) RETURN a, b, c"
+
+    val plan = (config getLogicalPlanFor query)._2
+
+    // Expected plan
+    // Since (a)--(b) has a lower cardinality estimate than (a)--(c) it should be selected first
+    val scanA = NodeByLabelScan("a", LabelName("A")(pos), Set.empty)(solved)
+    val expandB = Expand(scanA, "a", INCOMING, Seq.empty, "b", "rB", ExpandAll)(solved)
+    val selectionB = Selection(Seq(HasLabels(Variable("b")(pos), Seq(LabelName("B")(pos)))(pos)), expandB)(solved)
```

Comment:
Not very important, but `AstConstructionTestSupport` has helpful methods such as `hasLabels(v: String, l: String)` that creates some of the AST objects for you